### DISCUSSION TITLE
What about an Ungoogled Thorium?

### DISCUSSION THREAD

**Discussion by xplshn** [Eloston/ungoogled-chromium#2372] [General]
What about an Ungoogled Thorium? It seems Ungoogled Chromium really doesn't bring anything to the table, apart from being Chromium sans Google. On the other hand, Thorium is 38% faster than normal Chromium, its optimized, its compatible Chromium, its patches are well documented, its embeddable, because there's Thorium OS and they even mantain a Thorium Win7 edition.
Also, there's extension support, that works really well, its Chromium on steroids. Why don't we rebase Ungoogled Chromium to Ungoogled Thorium?

**Comment by PF4Public** [comment]
Looks like loop optimizations and Polly could give you 13.8 percent speedup, interesting.
Same hardware, same profile, ungoogled-chromium and Thorium.
![image](https://github.com/ungoogled-software/ungoogled-chromium/assets/10319700/f9769a85-2a1f-48e0-a591-eb5e7f7dcca9)
![image](https://github.com/ungoogled-software/ungoogled-chromium/assets/10319700/9db183e1-d45f-467e-859b-2c1f2dd7a0de)

**Reply by rany2** [reply]
is it something you could notice while using it? Browser benchmarks really are useless IMO

**Reply by rany2** [reply]
I bet loop optimizations are disabled to reduce binary size, so there is a tradeoff

**Reply by PF4Public** [reply]
> is it something you could notice while using it? Browser benchmarks really are useless IMO

This particular benchmark tests JS and html performance. Quote from its about page:

> Speedometer tests a browser's Web app responsiveness by timing simulated user interactions.
> 
> This benchmark simulates user actions for adding, completing, and removing to-do items using multiple examples in TodoMVC. Each example in TodoMVC implements the same todo application using DOM APIs in different ways. Some call DOM APIs directly from ECMAScript 5 (ES5), ECMASCript 2015 (ES6), ES6 transpiled to ES5, and Elm transpiled to ES5. Others use one of eleven popular JavaScript frameworks: React, React with Redux, Ember.js, Backbone.js, AngularJS, (new) Angular, Vue.js, jQuery, Preact, Inferno, and Flight. Many of these frameworks are used on the most popular websites in the world, such as Facebook and Twitter. The performance of these types of operations depends on the speed of the DOM APIs, the JavaScript engine, CSS style resolution, layout, and other technologies.

I do notice the difference in responsiveness between my Windows Vivaldi and Gentoo ungoogled-chromium on the same hardware. Last time I checked, Vivaldi was at 93 and ungoogled-chromium at 113. 

>I bet loop optimizations are disabled to reduce binary size, so there is a tradeoff

Google does not seem to care about Chromium binary size. Likely they do not want to deal with potential compiler bugs introduced by those flags.

**Reply by xplshn** [reply]
> is it something you could notice while using it? Browser benchmarks really are useless IMO

I use hardware that is by today's standards obsolete, Pentium 4 PCs, and an Intel Atom AIO. Its very noticeable when your hardware is just bad and every CPU cycle matters, also. When your PC is laggy and mostly frozen, you do notice that the browser unfreezes faster, I have been using both for like 6 months, ungoogled for sensitive and personal stuff, Thorium for most other things

**Reply by PF4Public** [reply]
@xplshn Your argument is totally correct: the older the hardware, the bigger the difference. On the other hand, older hardware lacks modern instructions, and it's becoming obsolete as software uses and requires more and more of those.

**Reply by xplshn** [reply]
> @xplshn Your argument is totally correct: the older the hardware, the bigger the difference. On the other hand, older hardware lacks modern instructions, and it's becoming obsolete as software uses and requires more and more of those.

Yeah, its obsolete, but its still functional hardware, also, sometimes you don't have another option. I haven't seen PCs newer than 2016 here, in Argentina. There are lots for sure, but not where I live, not even a single place, kiosks, schools, etc have modern hardware.

Anyways. I made this in the discussion page because it would be great to fork Thorium and add/adapt Ungoogled patches for it, or reimplementem if needed.

**Comment by xplshn** [comment]
Maybe if you virtualize an old CPU you could see a clearer difference, or by using the special releases with AVX instead of SSE

**Reply by PF4Public** [reply]
@xplshn The problem for now is that building Chromium is a significant task in its own and I doubt we can afford building several variations.

**Reply by xplshn** [reply]
But there's no need to, making the normal Thorium package plus the Ungoogled patches would be cool.
Its so sad that a lot of people that know about Thorium don't use it because its got the Chromium integrations enabled, so much wasted potential. 

https://github.com/Alex313031/thorium/issues?q=ungoogled+
https://github.com/Alex313031/thorium/issues/132

**Reply by PF4Public** [reply]
> But there's no need to, making the normal Thorium package plus the Ungoogled patches would be cool.

Anyone could take development scripts from this repository and prepare in a similar manner ungoogled Thorium. We barely have manpower to steer only ungooled-chromium itself.

**Reply by jstkdng** [reply]
And even if someone did that, I don't think it would have the same performance unless we adopt the same compiler flags thorium uses. Also, it seems thorium requires internet access to be able to build, making a lot of tools we already have unusable. Maybe if thorium packaged a ready to use .tar (like chromium) that requires no internet to build then it would be easy to adop into our existing build scripts.

**Comment by networkException** [comment]
Most of the thorium patches will be out of scope for the goals of this project.

There are two ways the two projects could interact:

1. Thorium adapts more ungoogled-chromium patches, which they are obviously free to do and are already doing to some capacity. We don't have any influence over that however
2. ungoogled-chromium packaging includes patches for the convenience of users wanting some of both. This would come with a hefty maintenance burden (packaging is already pretty resource limited (contributors and CI)) and not really feasible
